src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -463,15 +469,60 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             }
 
             return new FilteredCompletionModel(
-                GetHighlightedList(matchResults), index, filters,
+                GetHighlightedList(matchResults, filterText, highlightMatchingSpans, completionHelper), index, filters,
                 hardSelect ? UpdateSelectionHint.Selected : UpdateSelectionHint.SoftSelected,
                 centerSelection: true,
-                uniqueItem: moreThanOneMatchWithSamePriority ? null : bestMatchResult.GetValueOrDefault().VSCompletionItem);
+                uniqueItem: moreThanOneMatchWithSamePriority ? null : bestMatchResult.GetValueOrDefault().EditorCompletionItem);
         }
 
-        private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(List<MatchResult> matchResults)
-            => matchResults.SelectAsArray(matchResult =>
-            new CompletionItemWithHighlight(matchResult.VSCompletionItem, matchResult.HighlightedSpans));
+        private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(
```

Comment:
Previously, highlighting was done during the calculation of MatchResults and stored as a property on the MatchResult.  Now this is done when the model is being created (so just moving the computation slightly later) to avoid needing to store these spans on the MatchResult type.

The highlight calculation code was essentially moved unchanged, other than no longer storing on MatchResult

Reply:
can you clarify this though?  aren't the highlighted spans specific to each match?  so how can you store on the model and not the item?

Reply:
poke